comms/src/transports/tcp.rs

```diff
@@ -25,44 +25,42 @@ use crate::{
     transports::dns::{DnsResolverRef, SystemDnsResolver},
     utils::multiaddr::socketaddr_to_multiaddr,
 };
-use futures::{io::Error, ready, AsyncRead, AsyncWrite, Future, FutureExt, Stream};
+use futures::{ready, FutureExt};
 use multiaddr::Multiaddr;
 use std::{
+    future::Future,
     io,
     pin::Pin,
     sync::Arc,
     task::{Context, Poll},
-    time::Duration,
-};
-use tokio::{
-    io::{AsyncRead as TokioAsyncRead, AsyncWrite as TokioAsyncWrite},
-    net::{TcpListener, TcpStream},
 };
+use tokio::net::{TcpListener, TcpStream};
+use tokio_stream::Stream;
 
 /// Transport implementation for TCP
 #[derive(Clone)]
 pub struct TcpTransport {
-    recv_buffer_size: Option<usize>,
-    send_buffer_size: Option<usize>,
+    // recv_buffer_size: Option<usize>,
+    // send_buffer_size: Option<usize>,
     ttl: Option<u32>,
-    #[allow(clippy::option_option)]
-    keepalive: Option<Option<Duration>>,
+    // #[allow(clippy::option_option)]
+    // keepalive: Option<Option<Duration>>,
     nodelay: Option<bool>,
     dns_resolver: DnsResolverRef,
 }
 
 impl TcpTransport {
-    #[doc("Sets `SO_RCVBUF` i.e the size of the receive buffer.")]
-    setter_mut!(set_recv_buffer_size, recv_buffer_size, Option<usize>);
-
-    #[doc("Sets `SO_SNDBUF` i.e. the size of the send buffer.")]
-    setter_mut!(set_send_buffer_size, send_buffer_size, Option<usize>);
+    // #[doc("Sets `SO_RCVBUF` i.e the size of the receive buffer.")]
+    // setter_mut!(set_recv_buffer_size, recv_buffer_size, Option<usize>);
+    //
+    // #[doc("Sets `SO_SNDBUF` i.e. the size of the send buffer.")]
+    // setter_mut!(set_send_buffer_size, send_buffer_size, Option<usize>);
```

Comment:
remove?

Reply:
This (and all the others in this file) could come back - tokio doesn't support these socket options at the moment. I'd rather leave them in